docs/specs/src/hwp_4_formal.md

```diff
@@ -0,0 +1,534 @@
+
+![Operations and state changes produced by `delete_link` action](delete_link_ops.svg){#fig:delete_link_ops width=60%}
+
+\newpage
+
+#### Authority Election
+
+In the case of source chain entries (and actions), it is fairly obvious that the author who created them is the **author**itative source. But part of translating from a series of local chain states to a resilient global data store involves identifying which nodes in the network become the responsible authorities for holding which DHT content.
+
+Most existing DHT frameworks simply have nodes volunteer to hold specific content, and then use the DHT as a tracking layer to map content hashes to the nodes holding the content. But this allows content to disappear arbitrarily and also creates imbalanced dynamics between people who consume content and people who serve it (leechers & seeders). Since Holochain is designed to function more like a distributed database than a content distribution network, it needs to ensure resilience and permanence of data elements, as well as load balancing and reasonable performance, on a network where nodes are likely coming online and going offline frequently.
+
+As such, Holochain doesn't rely on nodes to volunteer to hold specific entries, but rather to volunteer aggregate capacity (e.g., holding 100MB of data rather than arbitrarily chosen entries). Authoring nodes are responsible for publishing entries from their local DHT instance to other nodes on the network (**authorities**) who will become responsible for serving that data.
+
+Like most DHT architectures, Holochain uses a "nearness" algorithm to compute the "distance" distance between the 256-bit Blake2b basis hash of the data or metadata to be stored and the 256-bit Ed25519 public key (network address) of nodes. Basically, it is the responsibility of the nodes *nearest* a basis hash to store data and metadata for it.
+
+Holochain's validating, graphing, gossiping DHT implementation is called **rrDHT**.
+
+rrDHT is designed with a few performance requirements/characteristics in mind.
+
+1. It must have a compact and computationally simple representational model for identifying which nodes are responsible for which content, and which nodes actually hold which content. (A **"world model"** of what is where.)
+2. It must have **lookup speeds** at least as fast as Kademlia's binary trees ( $\mathcal{O}(n \log n)$ ). Current testing shows an average of 3 hops/queries to reach an authority with the data.
+3. It must be **adjustable** to be both resilient and performant across many DHT compositional make-ups (reliability of nodes, different network topologies, high/low usage volumes, etc.)
+
+**World Model:** The network location space is a circle comprising of the range of unsigned 32-bit numbers, in which the location $0$ is adjacent to the location ${2^{32}}-1$.
+
+$$
+\begin{aligned}
+L &: \{l_0,\dots,l_{2^{32}-1}\} \\
+l_{2^{32}-1} &< l_0
```

Comment:
potentially simpler to just say it's the integers modulo 2^32, which is correct. Saying the last element is less than the first is formally problematic since it breaks the [transitive property of total order](https://en.wikipedia.org/wiki/Total_order).

```suggestion
L &: \mathbb{Z}_{2^32}
```